libraries/AP_InertialSensor/AP_InertialSensor_SCHA63T.cpp

```diff
@@ -0,0 +1,451 @@
+        ret_scha63t &= RegisterWrite(SCHA63T_UNO, RESCTRL, HW_RES);
+        // reset DUE write (0001h) to register 18h
+        ret_scha63t &= RegisterWrite(SCHA63T_DUE, RESCTRL, HW_RES);
+        // wait 25ms for non-volatile memory (NVM) read
+        hal.scheduler->delay(25);
+
+        // set DUE operation mode on (must be less than 1ms)
+        ret_scha63t &= RegisterWrite(SCHA63T_DUE, MODE, MODE_NORM);
+        ret_scha63t &= RegisterWrite(SCHA63T_DUE, MODE, MODE_NORM);
+        // set UNO operation mode on
+        ret_scha63t &= RegisterWrite(SCHA63T_UNO, MODE, MODE_NORM);
+        // wait 70ms initial startup
+        hal.scheduler->delay(50);
+
+        // set UNO configuration (data filter, flag filter)
+        ret_scha63t &= RegisterWrite(SCHA63T_UNO, G_FILT_DYN, G_FILT);
+        ret_scha63t &= RegisterWrite(SCHA63T_UNO, A_FILT_DYN, A_FILT);
+        // set DUE configuration (data filter, flag filter)
+        ret_scha63t &= RegisterWrite(SCHA63T_DUE, G_FILT_DYN, G_FILT);
+
+        // wait 45ms (adjust restart duration to 500ms)
+        hal.scheduler->delay(45);
+
+        if (!check_startup()) {
+            return false;    // check FAILED
+        }
+    }
+
+    // check ok
+    return true;
+}
+
+bool AP_InertialSensor_SCHA63T::check_startup()
+{
+    uint8_t val[4];
+    bool read_summary_error = false;
+
+    // wait 405ms (300Hz filter)
+    hal.scheduler->delay(405);
+
+    // start EOI = 1
+    ret_scha63t &= RegisterWrite(SCHA63T_UNO, RESCTRL, RES_EOI);
+    ret_scha63t &= RegisterWrite(SCHA63T_DUE, RESCTRL, RES_EOI);
+
+    // first read summary status
+    ret_scha63t &= RegisterRead(SCHA63T_UNO, S_SUM, val);
+    ret_scha63t &= RegisterRead(SCHA63T_DUE, S_SUM, val);
+    // 2.5ms or more
+    hal.scheduler->delay(3);
+
+    // second read summary status
+    ret_scha63t &= RegisterRead(SCHA63T_UNO, S_SUM, val);
+    ret_scha63t &= RegisterRead(SCHA63T_DUE, S_SUM, val);
+    // 2.5ms or more
+    hal.scheduler->delay(3);
+
+    // read summary status
+    ret_scha63t &= RegisterRead(SCHA63T_UNO, S_SUM, val);
+    // check UNO summary status
+    if (!((val[1] & 0x9e) && (val[2] & 0xda))) {
+        read_summary_error = true;
+    }
+    ret_scha63t &= RegisterRead(SCHA63T_DUE, S_SUM, val);
+    // check DUE summary status
+    if (!((val[1] & 0xf8) && (val[2] & 0x03))) {
+        read_summary_error = true;
+    }
+
+    // check error
+    if (read_summary_error) {
+        return false;
+    }
+
+    return true;
+}
+
+/*
+  read accel fifo
+ */
+void AP_InertialSensor_SCHA63T::read_accel(void)
+{
+    uint8_t rsp_accl_x[3];
+    uint8_t rsp_accl_y[3];
+    uint8_t rsp_accl_z[3];
+    uint8_t rsp_temper[3];
+
+    int16_t accel_x = 0;
+    int16_t accel_y = 0;
+    int16_t accel_z = 0;
+    int16_t uno_temp = 0;
+
+    // ACCL_X Cmd Send (This Response rsp_accl_x is Dust!!)
+    ret_scha63t &= RegisterRead(SCHA63T_UNO, ACC_X, rsp_accl_x);
+    // ACCL_Y Cmd Send + ACCL_X Response Receive
```

Comment:
can this be done as a single read block of all 3 registers?

Reply:
Yes.
you are right.